Sprint_6

DreamBreezz opened the pull request.

DreamBreezz committed:
add full project
add jacoco report

DreamBreezz committed:
nano fix (doesn't affect anything)

nikolay-977 commented on src/main/java/com/example/Lion.java:
```diff
@@ -5,8 +5,12 @@
 public class Lion {
 
     boolean hasMane;
+    Feline feline; // добавил инъекцию зависимости через конструктор
```
Необходимо исправить: Для всех полей следует указывать минимально необходимые модификаторы доступа

DreamBreezz replied:
done

nikolay-977 commented on src/main/java/com/example/Lion.java:
```diff
@@ -16,7 +20,7 @@ public Lion(String sex) throws Exception {
     }
 
-    Feline feline = new Feline();
+    //Feline feline = new Feline(); - убрал зависимость
```
Необходимо исправить: удалить лишний комментированный код

DreamBreezz replied:
done

nikolay-977 commented on src/test/java/com/example/LionDoesHaveManeTest.java:
```diff
@@ -0,0 +1,38 @@
+    public void lionMaleTest() throws Exception {
+        Lion lion = new Lion(lionSex, new Feline());
+            assertEquals(lion.doesHaveMane(), doesHaveMane);
+        throw new Exception();
```
Необходимо исправить: Проверки должны быть в конце тестовых методов. После проверок не должно быть выброса исключений.

nikolay-977 replied:
Можно улучшить: использовать специальные методы для проверки исключений. Примеры проверок exception:
https://www.baeldung.com/junit-assert-exception
https://www.baeldung.com/assertj-exception-assertion

DreamBreezz replied:
было сложно подружить проверку исключений, но у меня получилось. Полагаю, не самым изящным образом (не соблюдён принцип DRY), но хотя бы так

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Для оформления данных в виде таблице используй сочетание клавиш Ctrl + N или Cmd + N. Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

DreamBreezz committed:
fixes based on review:
- removed: commented code
- added: access modifiers in Lion.class
- updated: README.md
- updated: parameterized test LionDoesHaveManeTest.class with exceptions
- updated: new dependency in pom.xml (+ AssertJ)
- updated: new Jacoco report

nikolay-977 commented on src/test/java/com/example/LionDoesHaveManeTest.java:
```diff
@@ -0,0 +1,47 @@
+    // если самка, то гривы нет
+    // если пол не передан, ловим выброс исключения
+    @Test
+    public void lionMaleTest() throws Exception {
```
Можно улучшить: Не следует использовать конструкцию if-else при написании тестов. Это затрудняет читаемость кода и может вызвать путаницу в понимании. Лучше реализовать 2 отдельных кейса

DreamBreezz replied:
ирония в том, что я сначала так и сделал, а потом решил собрать проверки в один тест, через if-else :) вернул, как изначально сделал - проверку эксепшна в LionTest, заодно и AssertJ теперь не нужен.
Большое спасибо за ревью с подробными замечаниями, было весьма полезно! 👍

DreamBreezz committed:
fixes based on 2nd review:
- removed: AssertJ dependency
-- updated: pom.xml
-- updated: README.md
- moved: check exception moved from LionDoesHaveManeTest.java to LionTest.java
- updated: new Jacoco report
